File: backend/review-service/src/main/java/com/skillsync/review/entity/Review.java
package com.skillsync.review.entity;

import jakarta.persistence.*;
import java.time.LocalDateTime;

import com.skillsync.review.audit.Auditable;
import lombok.*;

@Entity
@Table(name = "reviews")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Review extends Auditable {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private Long mentorId;
    
    @Column(nullable = false)
    private Long learnerId;
    
    @Column(nullable = false)
    private Long sessionId;
    
    @Column(nullable = false)
    private Integer rating; // 1-5
    
    @Column(columnDefinition = "TEXT")
    private String comment;
    
    @Column(nullable = false)
    private Boolean isAnonymous = false;
    
    @Column(nullable = false)
    private LocalDateTime createdAt;
    
    @Column(nullable = false)
    private LocalDateTime updatedAt;
    
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend-react/src/types/review.ts
export interface Review {
  id: number;
  mentorId: number;
  learnerId: number;
  sessionId: number;
  rating: number;
  comment: string;
  createdAt: string;
  updatedAt: string;
  learnerName?: string;
  username?: string;
}

export interface SubmitReviewRequest {
  mentorId: number;
  sessionId: number;
  rating: number;
  comment: string;
}

export interface MentorRating {
  averageRating: number;
  totalReviews: number;
  ratingDistribution: Record<number, number>;
}
